Proposta alternativa de implementação

thdurante opened the pull request:
Proposta alternativa e consolidada (contendo as demais partes) conforme
discussão do grupo realizada em sala de aula.

thdurante committed:
Proposta alternativa de implementação - CONSOLIDAÇÃO

Proposta alternativa e consolidada (contendo as demais partes) conforme
discussão do grupo realizada em sala de aula.

thdurante committed:
Proposta alternativa de implementação parte 1

Proposta alternativa de implementação que o grupo discutiu em sala

kaiomagalhaes commented on openehr-rm-core/src/main/java/org/openehr/rm/formatters/JsonObject.java:
```diff
@@ -22,16 +22,28 @@ public void fromXML(String xml) {
+        String jsonFinal = "";
+        // -- talvez precise { } envolvendo o retorno do json, validar
+        idRaiz = obtemRaiz();
+        jsonFinal += "{";
```
coloque tudo em uma linha só, não é necessario adicionar cada charactere em uma linha, e não é necessário criar variavel já que é retornada +/- assim:
"{"+ buildJson(idRais) + "}"

kaiomagalhaes commented on openehr-rm-core/src/main/java/org/openehr/rm/formatters/JsonObject.java:
```diff
@@ -22,16 +22,28 @@ public void fromXML(String xml) {
 
     }
 
+    public void defineRaiz(int raiz) {
```
mude para definA, um método void deve estar no imperativo

thdurante replied:
Esse método veio do professor

kaiomagalhaes commented on openehr-rm-core/src/main/java/org/openehr/rm/formatters/JsonObject.java:
```diff
@@ -95,45 +151,95 @@ public int adicionaDvParsable(String codePhraseCharSet, String codePhraseLanguag
+                template = "'DvBoolean': {'value': #value}";
                 template = template.replaceAll("#value", String.valueOf(obtemValorLogico(idNodoGrafo, 0)));
-                template = template.replaceAll("'","\"");
+                template = template.replaceAll("'", "\"");
```
use aspas simples já que não utilizará regex

kaiomagalhaes commented on openehr-rm-core/src/main/java/org/openehr/rm/formatters/JsonObject.java:
```diff
@@ -95,45 +151,95 @@ public int adicionaDvParsable(String codePhraseCharSet, String codePhraseLanguag
-                template = template.replaceAll("#id", obtemTexto(idNodoGrafo,2));
-                template = template.replaceAll("#type", obtemTexto(idNodoGrafo, 3));
-                template = template.replaceAll("'","\"");
+                template = "'DvIdentifier': {'issuer': '#issuer', 'assigner': '#assigner', 'id': '#id', 'type': '#type'}";
```
use aspas simples já que não utilizará regex
linha 190 a 201

kaiomagalhaes committed:
Merge remote-tracking branch 'origin/master' into thdurante_branch

Conflicts:
	openehr-rm-core/src/main/java/org/openehr/rm/formatters/JsonObject.java

thdurante committed:
Solicitações de correções ajustadas

Arrumei o que o Kaio disse, não mudei a questão da nomenclatura do
método porque veio do professor e não mexi na questão das aspas, depois
a gente pode tirar
